### DISCUSSION TITLE
Github Pages Privacy error

### DISCUSSION THREAD

**Discussion by the-True-Blue** [community/community#32973] [Other Feature Feedback, Questions, & Ideas] [answered]
At random times, my website will reach a privacy error. I have enforced HTTPS on, it seems that all my links are through https and not http. So there should not be a mixed content issue. I am very unsure as to what is going on. I have to keep clearing cache on the browsers very frequently. It doesn't happen as much when you visit the site on a mobile phone so I'm not really sure if this is just an issue with chrome? I go to Firefox and everything is fine. It seems Chrome is the one that acts up the most. Then after a few hours it will go back to normal. But if I want the privacy error to disappear faster, then I have to clear cache.

This is my website here: tempest-digital.io

my repo: https://github.com/the-True-Blue/domportfolio

Just simply want to know what could be causing these random privacy errors to pop up. Whether its the links I have in my code or something else.

**Comment by airtower-luna** [comment, marked as answer]
What do you mean by "privacy error"? Please share the *exact and full* message you get, otherwise it's impossible to say what may be going on.

**Reply by the-True-Blue** [reply]
Apologies

What I meant by “privacy error” was that I keep getting “Your connection is
not private” on the page.

**Reply by airtower-luna** [reply]
Again, there should be more details on the error page. From the summary it's probably something related to (lack of) HTTPS and the certificate.

**Reply by the-True-Blue** [reply]
![E7C1EECA-0E4F-4170-9CE6-2810E617011E](https://user-images.githubusercontent.com/98621235/190505392-20d5c1bb-fa13-4ea1-92de-fb97c78e9b32.jpeg)

managed to find a picture of the exact error message.

says the common name is invalid.

**Reply by airtower-luna** [reply]
Now that's interesting. The error message indicates the certificate your browser receives doesn't match your domain. In general that can mean one of two things:

1. Your browser isn't talking to the right server for some reason (lots of different possible reasons!).
2. The server isn't configured correctly.

In your case it's easy to confirm it's the first option. A DNS query for your domain shows the problem:

```
$ drill tempest-digital.io
;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 47890
;; flags: qr rd ra ; QUERY: 1, ANSWER: 5, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; tempest-digital.io.	IN	A

;; ANSWER SECTION:
tempest-digital.io.	560	IN	A	185.199.110.153
tempest-digital.io.	560	IN	A	185.199.109.153
tempest-digital.io.	560	IN	A	192.124.249.8
tempest-digital.io.	560	IN	A	185.199.108.153
tempest-digital.io.	560	IN	A	185.199.111.153
```

One of those 5 IPs isn't one that belongs to GitHub, see the [custom domain documentation](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain) for the correct list (but it's not hard to see which one doesn't fit). Somewhat simplified, when looking up your domain the browser will pick one of those five at random when trying to connect to your site, so you have a 1 in 5 chance of getting that error. A working connection is reused for a while, so if you click around a bit after a working initial connection it'll work – until that connection gets closed.

Delete the bad IP address from your DNS config and it should work reliably. :slightly_smiling_face:

**Reply by the-True-Blue** [reply]
Ahh I see.

Before I remove the extra A record. I had screenshotted the record to fact check something with you. In the image attached, you’ll see that there’s actually two extra A records. I know that one of them with the name that says “sucuriip” is for activating a firewall on my site. I had bought a security package from GoDaddy where I can activate a firewall through 192.124.249.8 
and what will show up is [sucuri.net](http://sucuri.net/) which I believe is the firewall. 

Should I delete both records? I wasn’t sure if GitHub Pages provided firewall security automatically. I didn’t see any information in the official documentation detailing that. So I’m not sure if I should delete the firewall record too or is it just the A record with the name “@“?
![3D9C54A3-7DCF-416A-8B52-1847E5324B06](https://user-images.githubusercontent.com/98621235/190517601-3fea8678-4a4b-469e-8a07-3bba24aa022f.jpeg)

**Reply by airtower-luna** [reply]
> Should I delete both records? I wasn’t sure if GitHub Pages provided firewall security automatically.

Yes. You don't need a "firewall" there. GitHub takes care of the servers, and you only have a static site, no running service that could be exploited. I did take a brief look at the reverse DNS record for that IP and that `sucuri.net` site and thought "that looks fishy". :sweat_smile:

**Reply by the-True-Blue** [reply]
Ahh shoot. Tbh I’m gonna switch my domain from GoDaddy. I wish I done my research before choosing them. The shit I’ve heard is just horrible. And now selling fishy security packages? Yuck.

**Comment by the-True-Blue** [comment]
Apologies

What I meant by “privacy error” was that I keep getting “Your connection is
not private” on the page.

**Comment by jfullstackdev** [comment]
it seems that's only on your end if you have restrictions or your ISP, geographical location is important to consider too

**Reply by the-True-Blue** [reply]
![03B6FD5E-4F04-4337-A6C2-D81350BC87A2](https://user-images.githubusercontent.com/98621235/190505586-7f1ab2f6-6368-4416-a3ee-23872a0993ff.jpeg)

this is the full error message
